app/controllers/dashboard_controller.rb

```diff
@@ -359,10 +357,7 @@ def widget_add
     assert_privileges("dashboard_add")
     if params[:widget] # Make sure we got a widget in
       w = params[:widget].to_i
-      if @sb[:dashboards][@sb[:active_db]][:col3].length < @sb[:dashboards][@sb[:active_db]][:col1].length &&
-         @sb[:dashboards][@sb[:active_db]][:col3].length < @sb[:dashboards][@sb[:active_db]][:col2].length
-        @sb[:dashboards][@sb[:active_db]][:col3].insert(0, w)
-      elsif @sb[:dashboards][@sb[:active_db]][:col2].length < @sb[:dashboards][@sb[:active_db]][:col1].length
+      if @sb[:dashboards][@sb[:active_db]][:col2].length < @sb[:dashboards][@sb[:active_db]][:col1].length
         @sb[:dashboards][@sb[:active_db]][:col2].insert(0, w)
       else
         @sb[:dashboards][@sb[:active_db]][:col1].insert(0, w)
```

Comment:
Not for this PR, but this code can me made so much simpler:

```suggestion
      shortest_col = [:col1, :col2, :col3].min { |c| @sb[:dashboards][@sb[:active_db]][c].length }
      @sb[:dashboards][@sb[:active_db]][shortest_col].insert(0, w)
```